packages/wrapped-keys/src/lib/litActions.ts

```diff
@@ -49,43 +43,95 @@ const DEFAULT_GAS_PRICE = '50'; // in gwei
 
     const tx = {
         to: unsignedTransaction.toAddress,
-        value: ethers.utils.parseEther(unsignedTransaction.value),
+        value: ethers.utils.hexlify(ethers.utils.parseEther(unsignedTransaction.value)),
         chainId: unsignedTransaction.chainId,
-        gasPrice: ethers.utils.parseUnits(gasPrice, 'gwei'),
-        gasLimit,
         data: unsignedTransaction.dataHex,
         nonce,
     };
 
+    if (unsignedTransaction.gasPrice) {
+        tx.gasPrice = ethers.utils.parseUnits(unsignedTransaction.gasPrice, 'gwei');
+    } else {
+        tx.gasPrice = await Lit.Actions.runOnce(
+            { waitForResponse: true, name: 'gasPrice' },
+            async () => {
+                const rpcUrl = await Lit.Actions.getRpcUrl({ chain: unsignedTransaction.chain });
+                const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    
+                try {
+                    const gasPrice = await provider.getGasPrice();
+                    return ethers.utils.hexlify(gasPrice);
+                } catch (err) {
+                    const errorMessage = 'Error: When getting gas price- ' + err.message;
+                    Lit.Actions.setResponse({ response: errorMessage });
+                    return;
+                }
+            }
+        )
+    }
+
+    if (unsignedTransaction.gasLimit) {
+        tx.gasLimit = ethers.utils.hexlify(unsignedTransaction.gasLimit)
+    } else {
+        tx.gasLimit = await Lit.Actions.runOnce(
+            { waitForResponse: true, name: 'gasLimit' },
+            async () => {
+                const rpcUrl = await Lit.Actions.getRpcUrl({ chain: unsignedTransaction.chain });
+                const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+                
+                try {
+                    const gasLimit = await provider.estimateGas(tx);
+                    return ethers.utils.hexlify(gasLimit);
+                } catch (err) {
+                    const errorMessage = 'Error: When estimating gas- ' + err.message;
+                    Lit.Actions.setResponse({ response: errorMessage });
+                    return;
+                }
+            }
+        )
+    }
+
     console.log('tx');
     console.log(tx);
 
-    try {
-        const signedTx = await wallet.signTransaction(tx);
+    const signedTx = await Lit.Actions.runOnce(
+        { waitForResponse: true, name: "signedTx" },
+        async () => {
+            try {
+                return wallet.signTransaction(tx);
+            } catch (error) {
+                const errorMessage = 'Error: When signing transaction- ' + err.message;
+                Lit.Actions.setResponse({ response: errorMessage });
+                return;
+            }
```

Comment:
I'm not sure what happens when there's an error in `runOnce` maybe other nodes continue their execution since they aren't running this piece of code? But since it has the `waitForResponse: true` I think they'll time out.

I also think that if the node running this code throws an error then that won't be bubbled up to the user as a majority of the nodes will return "timeout error". We should test this: Make the `runOnce` throw an error and see what happens

Reply:
This might be worth consideration...

Reply:
I did:
```ts
if (unsignedTransaction.gasLimit) {
    tx.gasLimit = unsignedTransaction.gasLimit;
  } else {
    tx.gasLimit = await Lit.Actions.runOnce(
      { waitForResponse: true, name: 'gasLimit' },
      async () => {
        const rpcUrl = await Lit.Actions.getRpcUrl({
          chain: unsignedTransaction.chain,
        });
        const provider = new ethers.providers.JsonRpcProvider(rpcUrl);

        try {
          const gasLimit = await provider.estimateGas(tx);
          throw new Error('This is a fake error');
        } catch (err) {
          const errorMessage = 'Error: When estimating gas- ' + err.message;
          Lit.Actions.setResponse({ response: errorMessage });
          return;
        }
      }
    );
  }
```

and it seemed to have just skipped past the error and continue execution

Reply:
This is no longer done this way right since we exit if the node is not selected for decryption. Resolving